src/app/server/CommissioningWindowManager.cpp

```diff
@@ -150,7 +150,14 @@ void CommissioningWindowManager::HandleFailedAttempt(CHIP_ERROR err)
 #if CONFIG_NETWORK_LAYER_BLE
     mServer->GetBleLayerObject()->CloseAllBleConnections();
 #endif
-    if (mFailedCommissioningAttempts < kMaxFailedCommissioningAttempts)
+
+    bool closeWindow = false;
+    if (mAppDelegate != nullptr)
+    {
+        closeWindow = mAppDelegate->OnCommissioningSessionEstablishmentError(err);
```

Comment:
OK, this callsite very much does _not_ match the API documentation for when this function is called.  Which is correct?

Reply:
After the revert and doc updates, I believe the call to this function should now match with the docs. Let me know if it doesn't.